nucypher/utilities/prometheus/collector.py

```diff
@@ -330,13 +332,23 @@ def initialize(self, metrics_prefix: str, registry: CollectorRegistry) -> None:
             # use local event filter for initial data
             last_committed_period = self.contract_agent.get_last_committed_period(staker_address=self.staker_address)
             arg_filters = {'staker': self.staker_address, 'period': last_committed_period}
-            # check from the beginning - difficult to determine block number for previous period
-            # however, 'staker' and 'period' filters should help reduce load
-            initial_event_filter = self.contract_agent.contract.events[self.event_name].createFilter(
-                fromBlock=0, argument_filters=arg_filters)
-            events = initial_event_filter.get_all_entries()
-            for event in events:
-                self._event_occurred(event)
+            latest_block = self.contract_agent.blockchain.client.block_number
+            previous_period = self.contract_agent.get_current_period() - 1  # just in case
```

Comment:
I know it's not the scope of this PR, but I find weird how are we assuming here that `self.contract_agent` is a `StakingEscrowAgent`. Perhaps we should use a specific instance for it?

Reply:
`self.contract_agent` is obtained from the parent `EventMetricsCollector` class, which is a base class for all event collection including for other agents eg. WorkLock, PolicyManager.

Definitely open to improving the `contract_agent` to be more explicit if you have insight there - see https://github.com/derekpierre/nucypher/blob/prometheus-events-throttle/nucypher/utilities/prometheus/metrics.py#L204  for more context.